### PR TITLE
15 allow compiler to use asconfig

### LINKED ISSUE #15: Allow compiler to use asconfig
Allow compiler to use asconfig file located in the project generated by the project initialiser.

This issue is linked with https://github.com/massalabs/massa-sc-toolkit/issues/209

### PR DESCRIPTION
**Pull request by julienbrs**
close #15

### CODE REVIEW

**Comment by 0xSwapFeeder** [comment]
No sure to understand these changes, isn't asc using asconfig.json by default ?

**Comment by julienbrs** [comment]
> No sure to understand these changes, isn't asc using asconfig.json by default ?

Indeed asc use asconfig.json by default. However, in that file, we're using the AssemblyScript compiler programmatically with the asc.main method. I might be wrong, but it doesn't automatically load the asconfig.json. Thus, we need to manually populate arguments for the function

File: src/compiler.ts [hunk 1]
```diff
@@ -1,11 +1,37 @@
...
+
+    // Merge the targets and options present in the asconfig file with the options object
+    const target = asconfig.targets[mode || 'release'];
+    const targetOpts = Object.assign({}, asconfig.options, target);
+
+    // Merge command-line options with asconfig options
+    options = { ...targetOpts, ...options };
+
+    // // Creating protobuf files if it's in release mode
+    if ((options as any).transformer) {
```
Comment (gregLibert) [inline src/compiler.ts:27]: Can you use a type instead of any ?

File: src/compiler.ts [hunk 2]
```diff
@@ -1,11 +1,37 @@
...
+      console.log('Creating protobuf files');
+      argv.push('--transform');
+      argv.push((options as any).transformer[0]);
```
Comment (gregLibert) [inline src/compiler.ts:30]: same here

Comment (gregLibert) [inline src/compiler.ts:30]: What happens if you have more than one trasnformer ?

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: src/compiler.ts [hunk 3]
```diff
@@ -1,11 +1,49 @@
...
 import { join, basename } from 'path';
 import asc from 'assemblyscript/dist/asc.js';
 import yargs from 'yargs';
 import { hideBin } from 'yargs/helpers';
+import { APIOptions } from 'assemblyscript/dist/asc.js';
 
-async function compile(argv: string[], options: object = {}): Promise<boolean> {
+const asconfigPath = './asconfig.json';
+
+type CompilationOptions = APIOptions & {
```
Comment (gregLibert) [inline src/compiler.ts:11]: Why aren't you using compiler [Option](https://github.com/AssemblyScript/assemblyscript/blob/main/src/compiler.ts#L231)?

**Review by gregLibert — CHANGES_REQUESTED** [review]

**Review by gregLibert — APPROVED** [review]

**Pull request merged**